### PR TITLE
Bars for subjects or tags with little metrics in the dashboard would …

### LINKED ISSUE #5792: Bar charts are unreadable with a minimum (<3) number of metrics in that subject or with that label
**Describe the bug**
When selecting (or comparing) 2 or more dates, the bar charts are unreadable when there are a minimum (<3) number of metrics in that subject or with that label.

**To Reproduce**
Steps to reproduce the behaviour:
1. Select 2 or more number of dates (Settings - Number of dates)
2. Optional select "Time between dates"
3. See bar charts 

**Expected behaviour**
Shows all bar charts with a fixed width or a minimum (readable) width. 
Suggestion: use the width as currently used for 20 metrics.

**Screenshots**
image

### PR DESCRIPTION
**Pull request by fniessink**
…get too narrow to be legible. Fixes #5792.

### CODE REVIEW

File: components/frontend/src/dashboard/StatusBarChart.js [hunk 1]
```diff
@@ -19,7 +21,7 @@ export function StatusBarChart({ animate, colors, label, tooltip, summary, maxY,
         })
         return (
             <VictoryBar
-                barRatio={barRatio}
+                barWidth={(props) => ratio * (props.width - nrDates * 2 * props.style.padding - 2 * horizontalPadding) / nrDates}
```
Comment (wkoot) [inline components/frontend/src/dashboard/StatusBarChart.js]: Does this work the same on retina and regular displays?

File: components/frontend/src/dashboard/StatusBarChart.js [hunk 2]
```diff
@@ -33,18 +35,15 @@ export function StatusBarChart({ animate, colors, label, tooltip, summary, maxY,
     bars.reverse();
     colors.reverse();
     return (
-        <svg viewBox="0 0 400 400">
```
Comment (wkoot) [inline components/frontend/src/dashboard/StatusBarChart.js:36]: Wasn't this needed for tags under the cards, when the screen space is very narrow?

File: components/frontend/src/dashboard/StatusBarChart.js [hunk 3]
```diff
@@ -8,7 +8,9 @@ function nr_metrics_label(nr_metrics) {
 
 export function StatusBarChart({ animate, colors, label, tooltip, summary, maxY, style }) {
     const nrMetrics = sum(summary[Object.keys(summary)[0]])
-    const barRatio = Math.max(5, nrMetrics) / maxY
+    const nrDates = Object.keys(summary).length
+    const horizontalPadding = 50 + (7 - nrDates) * 10
```
Comment (wkoot) [inline components/frontend/src/dashboard/StatusBarChart.js]: Perhaps a comment to describe the calculation would help, seems like magic numbers to me

File: components/frontend/src/dashboard/StatusBarChart.js (hunk 1 above)
Comment (fniessink) [reply components/frontend/src/dashboard/StatusBarChart.js]: Unfortanately not. It looked great on my Retina display, but not a wide screen. Will fix this.

File: components/frontend/src/dashboard/StatusBarChart.js (hunk 2 above)
Comment (fniessink) [reply components/frontend/src/dashboard/StatusBarChart.js:36]: Made the graphs responsive, so no longer needed.

File: components/frontend/src/dashboard/StatusBarChart.js (hunk 3 above)
Comment (fniessink) [reply components/frontend/src/dashboard/StatusBarChart.js]: Code changed, but explanation added for the new calculation.

**Pull request merged**